src/Core/src/Platform/iOS/MauiView.cs

```diff
@@ -292,26 +292,45 @@ SafeAreaPadding GetAdjustedSafeAreaInsets()
 				{
 					// Get the keyboard frame and calculate its intersection with the current window
 					var window = this.Window;
-					
+
 					if (window != null && !_keyboardFrame.IsEmpty)
 					{
 						var windowFrame = window.Frame;
 						var keyboardIntersection = CGRect.Intersect(_keyboardFrame, windowFrame);
-						
+
 						// If keyboard is visible and intersects with window
 						if (!keyboardIntersection.IsEmpty)
 						{
-							// Calculate keyboard height in the window's coordinate system
-							var keyboardHeight = keyboardIntersection.Height;
-							
 							// For SafeAreaRegions.SoftInput: Always pad so content doesn't go under the keyboard
-							
 							// Bottom edge is most commonly affected by keyboard
 							var bottomEdgeRegion = safeAreaPage.GetSafeAreaRegionsForEdge(3); // 3 = bottom edge
 							if (SafeAreaEdges.IsSoftInput(bottomEdgeRegion))
 							{
 								// Use the larger of the current bottom safe area or the keyboard height
-								var adjustedBottom = Math.Max(baseSafeArea.Bottom, keyboardHeight);
+								// Get the input control's bottom Y in window coordinates
+								var inputBottomY = 0.0;
+								if (Window is not null)
+								{
+									var viewFrameInWindow = this.Superview?.ConvertRectToView(this.Frame, Window) ?? this.Frame;
+									inputBottomY = viewFrameInWindow.Y + viewFrameInWindow.Height;
+								}
+								var keyboardTopY = _keyboardFrame.Y;
+								var overlap = inputBottomY > keyboardTopY ? (inputBottomY - keyboardTopY) : 0.0;
+
+								// Prevent multiple nested SoftInput paddings by only applying if parent hasn't already applied it
+								var parentAppliedSoftInput = false;
```

Comment:
In theory if the code above is working shouldn't this code not be necessary? 

I tested the XAML you had sent me and if I comment out this code it seems to work fine

```XAML
    <Grid RowDefinitions="*,Auto"
          Padding="10"
          RowSpacing="8">
          
        <CollectionView/>

        <StackLayout SafeAreaEdges="SoftInput" Grid.Row="1">
            <ContentView Grid.Row="1"
                        SafeAreaEdges="All,All,All,SoftInput"
                        BackgroundColor="Purple">
                <Entry Background="Green"/>
            </ContentView>
        </StackLayout>
    </Grid>
```

<img width="415" height="867" alt="image" src="https://github.com/user-attachments/assets/eb0cc1ee-12c7-4513-9588-d6b0735953ca" />